src/primitives.rs

```diff
@@ -291,7 +291,10 @@ where
         version_map: &VersionMap,
         app_version: u16,
     ) -> VersionizeResult<()> {
-        let bytes_len = self.len() * std::mem::size_of::<T>();
+        let bytes_len = self
```

Comment:
We should also have a regression test for this fix :D

Reply:
Yes, this can be easily done in a unit-test.

Reply:
I will add a test for this.